code/lib/preview-web/src/PreviewWeb.test.ts

```diff
@@ -533,20 +530,48 @@ describe('PreviewWeb', () => {
                       `);
       });
 
-      it('emits but does not render exception if the play function throws', async () => {
-        const error = new Error('error');
-        componentOneExports.a.play.mockImplementationOnce(() => {
-          throw error;
+      describe('when `hidePlayFunctionExceptions` is set', () => {
+        it('emits but does not render exception if the play function throws', async () => {
+          const error = new Error('error');
+          componentOneExports.a.play.mockImplementationOnce(() => {
+            throw error;
+          });
+
+          getProjectAnnotations.mockReturnValueOnce({
+            ...projectAnnotations,
+            parameters: {
+              ...projectAnnotations.parameters,
+              hidePlayFunctionExceptions: true,
+            },
+          });
+
+          document.location.search = '?id=component-one--a';
+          const preview = await createAndRenderPreview();
+
+          expect(mockChannel.emit).toHaveBeenCalledWith(
+            PLAY_FUNCTION_THREW_EXCEPTION,
+            serializeError(error)
+          );
+          expect(preview.view.showErrorDisplay).not.toHaveBeenCalled();
```

Comment:
I would like to check for `STORY_THREW_EXCEPTION` here too, because that's an actual API whereas `showErrorDisplay` is an implementation detail.

Reply:
@ghengeveld do you think we should be throwing `STORY_THREW_EXCEPTION` in addition to the `PLAY_FUNCTION_THREW_EXCEPTION`? I wonder if in fact we should expect the opposite?

> whereas showErrorDisplay is an implementation detail

Hmm, maybe the opposite -- maybe we don't want to change the API in this PR, simply the implementation details (what does the user see?)

Reply:
I was suggesting to add this:

```js
expect(mockChannel.emit).not.toHaveBeenCalledWith(
  STORY_THREW_EXCEPTION,
	serializeError(error)
);
```

And do the opposite (where we DO emit this event, and DO show the error) below